OCPBUGS-5428: Add missing SDK extensions descriptions (#12431)
openshift/console: frontend/packages/console-dynamic-plugin-sdk/docs/console-extensions.md

### Summary 

> Allows to trigger a specific action when a specific Prometheus alert, based on its rule.name value, is observed by Console.

Author: definitely, as most of these descriptions where provided by non-native speakers I wanted to ask you for a review :)

### Properties

| Name | Value Type | Optional | Description |
| ---- | ---------- | -------- | ----------- |
| `alert` | `string` | no | Alert name as defined by `alert.rule.name` property |
| `text` | `string` | no | Action text |
| `action` | `CodeRef<(alert: Alert, launchModal: LaunchModal) => void>` | no | Function to perform side effect |

---

## `console.catalog/item-filter`

### Summary 

> Allows plugins to contribute a handler that can filter specific catalog items.<br/>For example plugin can contribute a filter that filters helm charts from specific provider.

Author: hmm, yeah I see your point but, semantically speaking the definition is right and makes sense. right ?

> Allows plugins to contribute a handler that can filter specific catalog items.<br/>For example plugin can contribute a filter that filters helm charts from specific provider.

Author: Would leave as you sugggested

### Summary 

> Allows plugins to contribute a new type of catalog item. For example Helm plugin can define<br/>a new catalog item type as HelmCharts that it wants to contribute to the Developer Catalog.

Author: HelmCharts since we are referencing a CRD/resource kind

### Summary 

> This extension can be used to contribute a navigation item that points to a namespaced resource details page<br/>The K8s model of that resource can be used to define the nav item.

Author: the first one should be good 👍

### Summary 

> This extension can be used to define a new section of nav items in the navigation.

Author: yep, this definitely makes more sense 👍

### Summary 

> This extension can be used to add a separator between nav items of nav sections in the navigation.

Author: > Same as comment above. Maybe tab or field at the end here. Not sure. I am leaning toward field.
not sure what you mean

| Name | Value Type | Optional | Description |
| ---- | ---------- | -------- | ----------- |
| `CSI` | `ProvisionerDetails` | yes | Constainer Storage Interface provisioner type |
| `OTHERS` | `ProvisionerDetails` | yes | Others provisioner type |

---

## `console.storage-provider`

### Summary 

> Allows contributing new storage provider to select when attaching a storage along with provider specific component.

Reviewer: I think this needs wordsmithing a bit for clarity. I am not sure I captured it correctly:
```suggestion
This extension can be used to contribute a new storage provider to select when attaching storage and a provider specific component.
```

### Summary 

> Allows to add a tab on the resource details page.

Reviewer: ```suggestion
This extension can be used to add a tab on the resource details page.
```

### Summary 

> Registers a listener function which receives telemetry events.These events include user identification,<br/>page navigation, and other application specific events. The listener may use this data for reporting and analytics purposes.

Reviewer: ```suggestion
This component can be used to register a listener function receiving telemetry events. These events include user identification,<br/>page navigation, and other application specific events. The listener may use this data for reporting and analytics purposes.
```

| Name | Value Type | Optional | Description |
| ---- | ---------- | -------- | ----------- |
| `provides` | `CodeRef<RelationshipProviderProvides>` | no | use to determine if a connection can be created between the source and target node |
| `tooltip` | `string` | no | tooltip to show when connector operation is hovering over the drop target ex: "Create a Visual Connector" |
| `create` | `CodeRef<RelationshipProviderCreate>` | no | callback to execute when connector is drop over target node to create a connection |
| `priority` | `number` | no | priority for relationship, higher will be preferred in case of multiple |

---

## `console.user-preference/group`

### Summary 

> Allows to add a group on the console user preferences page. Which will appears as a vertical tab option on the console user-preferences page.

Reviewer: ```suggestion
This extension can be used to add a group on the console user preferences page. It will appear as a vertical tab option on the console user-preferences page.
```

### Summary 

> Allows to add an item to the user preferences group on the console user preferences page.

Reviewer: ```suggestion
This extension can be used to add an item to the user preferences group on the console user preferences page.
```

### Summary 

> Allows plugins to contribute an add action item to the add page of developer perspective.<br/>For example Serverless plugin can add a new action item for adding serverless functions<br/>to the add page of developer console.

Reviewer: ```suggestion
This extension allows plugins to contribute an add action item to the add page of developer perspective.<br/>For example, a Serverless plugin can add a new action item for adding serverless functions<br/>to the add page of developer console.
```

### Summary 

> Allows plugins to group together multiple add actions in the add page of developer console.<br/>For example, a serverless plugin can contribute a Serverless group and multiple add actions.<br/>All these actions will be grouped together in the add action page based on this extension definition.

Reviewer: ```suggestion
This extension allows plugins to group together multiple add actions in the add page of developer console.<br/>For example, a serverless plugin can contribute a Serverless group and multiple add actions.<br/>All these actions will be grouped together in the add action page based on this extension definition.
```

### Summary 

> Allows specifying extra build environment variable fields under the builder image selector<br/>in the dev console git import form. The fields when set will override environment variables<br/>of the same name in the build section.

Reviewer: ```suggestion
This extension can be used to specify extra build environment variable fields under the builder image selector<br/>in the dev console git import form. When set, the fields will override environment variables<br/>of the same name in the build section.
```
